libmpv/csharp/Form1.cs

```diff
@@ -0,0 +1,167 @@
+        private delegate int MpvInitialize(IntPtr mpvHandle);
+        private MpvInitialize _mpvInitialize;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvCommand(IntPtr mpvHandle, [MarshalAs(UnmanagedType.LPArray)] string[] args);
+        private MpvCommand _mpvCommand;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvTerminateDestroy(IntPtr mpvHandle);
+        private MpvTerminateDestroy _mpvTerminateDestroy;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetOption(IntPtr mpvHandle, byte[] name, int format, ref uint data);
+        private MpvSetOption _mpvSetOption;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetOptionString(IntPtr mpvHandle, byte[] name, byte[] value);
+        private MpvSetOptionString _mpvSetOptionString;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvGetPropertystring(IntPtr mpvHandle, byte[] name, int format, ref IntPtr data);
+        private MpvGetPropertystring _mpvGetPropertyString;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetProperty(IntPtr mpvHandle, byte[] name, int format, ref byte[] data);
+        private MpvSetProperty _mpvSetProperty;
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private object GetDllType(Type type, string name)
+        {
+            IntPtr address = GetProcAddress(_libMpvDll, name);
+            if (address != IntPtr.Zero)
+                return Marshal.GetDelegateForFunctionPointer(address, type);
+            return null;
+        }
+
+        private void LoadMpvDynamic()
+        {
+            _libMpvDll = LoadLibrary("mpv-1.dll"); // The dll is included in the DEV builds by lachs0r: https://mpv.srsfckn.biz/
+            _mpvCreate = (MpvCreate)GetDllType(typeof(MpvCreate), "mpv_create");
+            _mpvInitialize = (MpvInitialize)GetDllType(typeof(MpvInitialize), "mpv_initialize");
+            _mpvTerminateDestroy = (MpvTerminateDestroy)GetDllType(typeof(MpvTerminateDestroy), "mpv_terminate_destroy");
+            _mpvCommand = (MpvCommand)GetDllType(typeof(MpvCommand), "mpv_command");
+            _mpvSetOption = (MpvSetOption)GetDllType(typeof(MpvSetOption), "mpv_set_option");
+            _mpvSetOptionString = (MpvSetOptionString)GetDllType(typeof(MpvSetOptionString), "mpv_set_option_string");
+            _mpvGetPropertyString = (MpvGetPropertystring)GetDllType(typeof(MpvGetPropertystring), "mpv_get_property");
+            _mpvSetProperty = (MpvSetProperty)GetDllType(typeof(MpvSetProperty), "mpv_set_property");
+        }
+
+        public void Pause()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return;
+
+            var bytes = Encoding.UTF8.GetBytes("yes\0");
```

Comment:
It seems a bit fragile to have to call GetBytes and add the trailing '\0' every time. Maybe there should be a helper function for creating nul-terminated UTF-8 strings.